master/internal/api_workspace.go

```diff
@@ -615,7 +616,49 @@ func (a *apiServer) UnpinWorkspace(
 }
 
 func (a *apiServer) ListRPsBoundToWorkspace(
-	ctx context.Context, request *apiv1.ListRPsBoundToWorkspaceRequest,
+	ctx context.Context, req *apiv1.ListRPsBoundToWorkspaceRequest,
 ) (*apiv1.ListRPsBoundToWorkspaceResponse, error) {
-	return &apiv1.ListRPsBoundToWorkspaceResponse{}, nil
+	curUser, _, err := grpcutil.GetUser(ctx)
+	if err != nil {
+		return nil, err
+	}
+	err = workspace.AuthZProvider.Get().CanGetWorkspaceID(
+		ctx, *curUser, req.WorkspaceId,
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	rpWorkspaceBindings, pagination, err := db.ReadRPsBoundToWorkspace(
+		ctx, req.WorkspaceId, req.Offset, req.Limit,
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	boundRPs := []string{}
+	for _, rpWorkspaceBinding := range rpWorkspaceBindings {
+		boundRPs = append(boundRPs, rpWorkspaceBinding.PoolName)
+	}
+
+	// Add unbound RPs.
```

Comment:
for pagination, I think that it's ok to overwrite some of the attributes to reflect the total number of pools available (bound + unbound). Looking at the code, I'm not quite sure how to interpret the logic, but if we can change it so that you include the unbound in the vars, I think that will work

Reply:
Added unbound pools to the paginated query, so it can be more robust than a custom logic.